### PR TITLE
feat: add linting to project

### LINKED ISSUE #12: Adicionar linter ao projeto

### PR DESCRIPTION
**Pull request by frattezi**
**Why this PR is necessary, what does it do**

This PR installs and configures prettier as the linter/formater alongside the default eslint configuration. In general, I've copied the PodCodar WebApp configuration and tested it. Also, it adds a new automated quality gate workflow using GA, basically checking build and lint for the code on every MR.

**References**
- Closes #12

### CODE REVIEW

File: .github/workflows/merge-request.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          node-version: 16.x
+      
+      - name: Install dependencies
+        run: yarn
+      
+      - name: Lint code
+        run: yarn lint
+      
+      - name: Check build
+        run: yarn build
```
Comment (marco-souza) [inline .github/workflows/merge-request.yaml:26]: missing empty line

File: pages/api/hello.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
 // Next.js API route support: https://nextjs.org/docs/api-routes/introduction
-import type { NextApiRequest, NextApiResponse } from 'next'
+import type { NextApiRequest, NextApiResponse } from 'next';
```
Comment (marco-souza) [inline pages/api/hello.ts:2]: we might wanna remove this into another PR

**Review by marco-souza — APPROVED** [review]
Looks good, left a couple of nitpick comments, but nothing critical.

File: .github/workflows/merge-request.yaml (hunk 1 above)
Comment (frattezi) [reply .github/workflows/merge-request.yaml:26]: lol my bad

File: pages/api/hello.ts (hunk 2 above)
Comment (frattezi) [reply pages/api/hello.ts:2]: Totally, I will deal with the API in my next task.

File: .github/workflows/merge-request.yaml (hunk 1 above)
Comment (frattezi) [reply .github/workflows/merge-request.yaml]: Done

**Review by marco-souza — APPROVED** [review]

**Pull request merged**